app/components/FieldString/FieldStringEdit.jsx

```diff
@@ -133,31 +211,217 @@ export default class FieldStringEdit extends React.Component {
+            ...valueArray.slice(index + 1)
+          ]
+
+          onChange(newValueArray)
+
+          this.cursor = {index: index + 1, selectionRange: [0, 0]}
+        }
+
+        return
+
+      case 'Backspace':
+        if (
+          selectionStart === 0 &&
+          selectionEnd === 0 &&
+          index < valueArray.length &&
+          index > 0 &&
+          typeof onChange === 'function'
+        ) {
+          // Necessary – otherwise removes last char of previous item.
+          e.preventDefault()
+
+          // Concatenate the item with the previous one.
+          const prevItem = valueArray[index - 1] || ''
+          const currItem = valueArray[index] || ''
+          const newValueArray = [
+            ...valueArray.slice(0, index - 1),
+            prevItem + currItem,
+            ...valueArray.slice(index + 1)
+          ]
+
+          onChange(newValueArray)
+
+          this.cursor = {
+            index: index - 1,
+            selectionRange: [prevItem.length, prevItem.length]
+          }
+        } else if (index === valueArray.length) {
+          e.preventDefault()
+
+          this.setCursor({index: index - 1})
+        }
+
+        return
+
+      case 'Delete':
+        if (
+          selectionStart === value.length &&
+          selectionEnd === value.length &&
+          index < valueArray.length - 1
+        ) {
+          // Necessary – otherwise removes first char of next item.
+          e.preventDefault()
+
+          // Concatenate the item with the next one.
+          const currItem = valueArray[index] || ''
+          const nextItem = valueArray[index + 1] || ''
+          const newValueArray = [
+            ...valueArray.slice(0, index),
+            currItem + nextItem,
+            ...valueArray.slice(index + 2)
+          ]
+
+          onChange(newValueArray)
+
+          this.cursor = {
+            index,
+            selectionRange: [currItem.length, currItem.length]
+          }
+        }
+
+        return
+
+      case 'ArrowUp': {
+        if (index > 0 && selectionStart === 0 && selectionEnd === 0) {
+          e.preventDefault()
+
+          this.setCursor({
+            index: index - 1,
+            selectionRange: [0, 0]
+          })
+        }
+
+        return
+      }
+
+      case 'ArrowDown': {
+        if (
+          index < valueArray.length &&
+          selectionStart === value.length &&
+          selectionEnd === value.length
+        ) {
+          e.preventDefault()
+
+          this.setCursor({
+            index: index + 1,
+            selectionRange: [0, 0]
+          })
+        }
 
-    if (typeof onChange === 'function') {
-      onChange.call(this, sanitisedValue, options)
+        return
+      }
     }
   }
 
+  handleListInputChange(index, e) {
+    const {onChange, value: oldValueArray} = this.props
+
+    if (typeof onChange !== 'function') return
+
+    const newValueArray = Array.isArray(oldValueArray)
+      ? oldValueArray.slice()
```

Comment:
Are you calling `slice` with no arguments to get a new array? If so, can we use `Array.from` instead?

Reply:
Fixed